cmd/zpool/zpool_iter.c

```diff
@@ -321,6 +321,83 @@ for_each_vdev(zpool_handle_t *zhp, pool_vdev_iter_f func, void *data)
 	return (for_each_vdev_cb(zhp, nvroot, func, data));
 }
 
+/*
+ * Process a line of command output
+ *
+ * When running 'zpool iostat|status -c' the lines of output can either be
+ * in the form of:
+ *
+ * 	column_name=value
+ *
+ * Or just:
+ *
+ * 	value
+ *
+ * Process the column_name (if any) and value.
+ *
+ * Returns 1 if line was processed, and there are more lines can still be
+ * processed.
+ *
+ * Returns 0 if this was the last line to process, or error.
+ */
+static int
+vdev_process_cmd_output(vdev_cmd_data_t *data, char *line)
+{
+	char *col = NULL;
+	char *val = line;
+	char *equals;
+
+	if (!line)
+		return (0);
+
+	equals = strchr(line, '=');
+
+	if (equals) {
+		/*
+		 * We have a 'column=value' type line.  Split it into the
+		 * column and value strings by turning the '=' into a '\0'.
+		 */
+		*equals = '\0';
+		col = line;
+		val = equals + 1;
+	} else {
+		val = line;
+	}
+
+	if (val != NULL) {
+		data->lines = realloc(data->lines,
+		    (data->lines_cnt + 1) * sizeof (*data->lines));
```

Comment:
Using `realloc()` in this way results in a memory leak on failure.  When NULL is returned we'll overwrite the pointer to the allocated block and it won't be freed by `realloc()`.